src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -81,22 +78,27 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
               IndexLogEntry.schemaString(schema),
               numBuckets)),
           Content(path.toString, Seq()),
-          Source(SparkPlan(sourcePlanProperties), Seq(Hdfs(sourceDataProperties))),
+          Source(
+            SparkPlan(sourcePlanProperties), Seq(Hdfs(sourceDataProperties)), relations),
           Map())
 
       case None => throw HyperspaceException("Invalid plan for creating an index.")
     }
   }
 
-  protected def sourceFiles(df: DataFrame): Seq[String] =
+  protected def sourceRelations(df: DataFrame): Seq[RelationMetadataEntry] =
```

Comment:
We support covering index from a single source relation. Should we just return `RelationMetadataEntry`. Multiple relations are needed for indexed views, but they cannot work if we remove the serialized df anyway.
Thoughts? @apoorvedave1 / @pirz / @rapoth

Reply:
This is not relevant after https://github.com/microsoft/hyperspace/pull/99/files#r463881967